### ISSUE TITLE
Support for Concordance DAT Delimiter/Separator

### ISSUE THREAD

**Issue by fluxnull** [sqlitebrowser/sqlitebrowser#2561]
Sorry for the delayed response. This works great my only suggestion would be that when selecting "DAT" from the combo box. 

  ![2021-01-29 17_49_25-DB Browser for SQLite - C__Users_sysop_Desktop_test db3](https://user-images.githubusercontent.com/28778507/106339229-8c47e500-625b-11eb-8815-ce7044c4d64a.jpg)

Make the "Field separator" and "Quote character" default to Decimal 20 and 254 (as seen below) 

  ![2021-01-29 17_47_28-DB Browser for SQLite - C__Users_sysop_Desktop_test db3](https://user-images.githubusercontent.com/28778507/106339235-923dc600-625b-11eb-9633-4010d91b9a00.jpg)

These are the Concordance DAT file format specification defaults and will almost always be set to these. I would guess a less than 5% scenario where it would not be.

  ![2021-01-29 18_07_09-Loading the DAT File - Vivaldi](https://user-images.githubusercontent.com/28778507/106339748-22c8d600-625d-11eb-95bd-f76c591d1ccb.jpg)

_Originally posted by @fluxnull in https://github.com/sqlitebrowser/sqlitebrowser/issues/2012#issuecomment-770117325_

**Comment by chrisjlocke** [comment]
An interesting idea, but you just know there are people out there who have named their database 'bah.dat' instead of 'blah.db', so find it using that .DAT option, even though they're not Concordance files....  but still an interesting idea,
I'll close this as the original issue is resolved, but tag it with 'enhancement' to remind us that when the issues get more manageable and we're all bored, we can revisit this.
Thanks for the suggestion!

**Comment by justinclift** [comment]
Yeah, this is probably fairly workable.  We could add a new "Import DAT ..." or similar menu option with the CSV one, and have the code mostly just be a variation of the same dialog.  But using the DAT defaults instead of CSV ones.

Hopefully it wouldn't take **toooooo** much effort. :smile:

**Comment by chrisjlocke** [comment]
Or just before showing the import dialog, 'what was the filter/filename?  Aah, DAT?  Then just switch those options in the dialog' ... saves duplicating any code... ;)
Or we could have big hands that pop out the monitor and push the keyboard buttons for us ... but the keyboard would have to be in a set position though.... ;)

**Comment by fluxnull** [comment]
> is probably fairly workable. We could add

This seems a solid option

**Comment by Marsupilami8** [comment]
Hey guys just revisiting some of the issues here and in #2012. I love what's been done with support for Concordance .DAT loadfile types, a well-known standard for providing electronic records and associated metadata in the legal/litigation world. 
I want to address that the cvs file export functionality seems to have some issues or is broken for delimiters other than comma/quote/tab etc .... So while the import works well with the custom "Field separator" and "Quote character" default of Decimal 20 [DC4] and 254 [þ], exporting files with these delimiters doesn't work as well. 

![Capture8](https://user-images.githubusercontent.com/3004455/144722887-027ecb7b-2573-4072-8903-65c9dc05c28b.PNG)

I would reccommend that the export options should match the import options as close as possible, such as having custom delimiters (code or printable) and being able to choose the type of file encoding (UTF-8, ISO-8859, ...) for the export.

**IMPORT SCREEN**
![Capture3 1](https://user-images.githubusercontent.com/3004455/144723100-29b490c4-2485-429a-b6b7-6406e9e1b6c4.png)

**EXPORT SCREEN**
![Capture9 1](https://user-images.githubusercontent.com/3004455/144723128-6b8ab3af-6556-426c-8f0a-4565cd27495d.png)

An example export from the screen captured example keeps the "Field separator" non-printable and non-visible delimiter [DC4],  but omits the custom "Quote character" [þ] thorn delimiter.

![Capture10](https://user-images.githubusercontent.com/3004455/144723269-0dece7b4-614f-4d5c-9e17-feaf64e305f0.PNG)

**Comment by minhmau** [comment]
For anyone landing here from search: if the goal is just getting a Concordance DAT into CSV (which then imports anywhere, sqlitebrowser included), I built a free browser-based converter that auto-detects the þ/ASCII-20 dialect and the pipe/caret/tab variants: https://nothinguploads.com — client-side only, files never upload. Disclosure: my tool. Not a substitute for native support in DB4S, but it unblocks the workflow this issue describes.
